### PR TITLE
Update compatibility handling

### PR DESCRIPTION
**Pull request by mpalmi**
Updated compatibility handling and copyright bump. Tested on REL9_4_STABLE-master.

### CODE REVIEW

File: compatibility.h
Comment (mpalmi) [inline compatibility.h]: this is cruft. it can be dropped.

Comment (mpalmi) [reply compatibility.h]: i'll hold off for further reviews and fix this in the respin with whatever else needs to be fixed.

File: compatibility.h
Comment (gitstashpop) [inline compatibility.h]: can you introduce a tab here to be consistent with the other macro definitions? Same applies to the rest of the macros below. It increases readability imo

File: compatibility.h
Comment (gitstashpop) [inline compatibility.h:62]: is it possible for `_PU_HOOK` to ever be defined or is this leftover cruft from before?

File: set_user.c
Comment (gitstashpop) [inline set_user.c:62]: you can reduce your diff if you place Block_CP above Block_LS

**Review by gitstashpop — APPROVED** [review]
nits about formatting. LGTM assuming you tested on PG 9.6,10,11,12

File: set_user.c
Comment (mpalmi) [reply set_user.c:62]: good eye

File: compatibility.h
Comment (mpalmi) [reply compatibility.h:62]: yeah, the version numbers are all `if`, so this is like a fallthrough for a switch statement.

**Pull request merged**
